docker/pypi/wmagent/run.sh

```diff
@@ -0,0 +1,683 @@
+        elif [[ $service = "rucio" ]]
+        then
+            [[ -d $WMA_CONFIG_DIR/$service/etc ]] || mkdir -p $WMA_CONFIG_DIR/$service/etc ; let errVal+=$?
+            cp -f $WMA_DEPLOY_DIR/etc/${config} $WMA_CONFIG_DIR/$service/etc/ ; let errVal+=$?
+        else
+            [[ -d $WMA_CONFIG_DIR/$service ]] || mkdir -p $WMA_CONFIG_DIR/$service ; let errVal+=$?
+            cp -f $WMA_DEPLOY_DIR/etc/${config} $WMA_CONFIG_DIR/$service/ ; let errVal+=$?
+        fi
+        [[ $errVal -eq 0 ]] && echo $WMA_BUILD_ID > $WMA_CONFIG_DIR/$service/.dockerInit
+    done
+
+    # Check if the host has a basic WMAgent.secrets file and copy a template if missing
+    # NOTE: Here we never overwrite any existing WMAGent.secrerts file: We follow:
+    #       * Check if there is any at the host, and if so, is it a blank template or a fully configured one
+    #       * In case we find a legit WMAgent.secrets file we set the .dockerInit and move on
+    #       * In case we need to copy a brand new template (based on the agent tape - test/prod)
+    #         or a blank one found at the host we halt without updating the .dockerInit file
+    #         and we ask the user to examine/update the file.
+    #       (Re)Initialisation should never pass beyond that step unless properly
+    #       configured WMAgent.secrets file being provided at the host.
+    echo "$FUNCNAME: Initialise WMAgent.secrets"
+    _init_valid $WMA_HOSTADMIN_DIR || {
+        if [[ ! -f $WMA_HOSTADMIN_DIR/WMAgent.secrets ]]; then
+            # NOTE: we consider production templates for relval agents and testbed templates for dev- agents
+            local agentType=${TEAMNAME%%-*}
+            agentType=${agentType/relval*/production}
+            agentType=${agentType/dev*/testbed}
+            echo "$FUNCNAME: copying $WMA_DEPLOY_DIR/etc/WMAgent.$agentType to $WMA_HOSTADMIN_DIR/WMAgent.secrets"
+            cp -f $WMA_DEPLOY_DIR/etc/WMAgent.$agentType $WMA_HOSTADMIN_DIR/WMAgent.secrets
+        fi
+        echo "$FUNCNAME: checking $WMA_HOSTADMIN_DIR/WMAgent.secrets"
+        if (_parse_wmasecrets $WMA_HOSTADMIN_DIR/WMAgent.secrets); then
+            echo $WMA_BUILD_ID > $WMA_HOSTADMIN_DIR/.dockerInit
+        else
+            echo "ERROR: We found a blank WMAgent.secrets file template at the current host!"
+            echo "ERROR: Please update it properly before reinitialising the WMagent container!"
+            return $(false)
+        fi
+    }
+
+    echo "Done: $stepMsg"
+    echo "-------------------------------------------------------"
+}
+
+check_wmasecrets(){
+    # Check if the the current WMAgent.secrets file is the same as the one from the latest agent inititialisation
+    echo "$FUNCNAME: Checking for changes in the WMAgent.secrets file"
+    touch $WMA_HOSTADMIN_DIR/.WMAgent.secrets.md5
+    if (md5sum --quiet -c $WMA_HOSTADMIN_DIR/.WMAgent.secrets.md5); then
```

Comment:
I am not sure I understand this line. What I read is that the checksum is read and checked from `.WMAgent.secrets.md5`, but where does the original secrets file checksum come from? Where is their comparison?

Reply:
This is the actual comparison. While  the creation of the sum itself happens during the `deploy_to_container` step.

Reply:
Ohh, the `touch` command above will not do anything if a .WMAgent.secrets.md5 file already exist. It's now clear to me. Thanks